### PR TITLE
Patrick Duvall

### PR DESCRIPTION
**Pull request by Patrick-Duvall**

### CODE REVIEW

File: lib/charachter.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class Character
+  attr_reader :name, :actor, :salary
+
+  def initialize(hash)
+    @name = hash[:name]
+    @actor = hash[:actor]
+    @salary = hash[:salary]
+  end
+  
+end
```
Comment (mstang15) [inline lib/charachter.rb]: 💯

File: lib/network.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    @shows = []
+  end
+
+  def add_show(show)
+    @shows << show
+  end
+
+  def highest_paid_actor
+    all_charachters = @shows.map(&:characters).flatten
+    highest_paid_charachter = all_charachters.max_by{|char| char.salary}
```
Comment (mstang15) [inline lib/network.rb]: Great job here! Good use of max_by!

File: lib/network.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    highest_paid_charachter.actor
+  end
+
+  def payroll
+      all_charachters = @shows.map(&:characters).flatten
```
Comment (mstang15) [inline lib/network.rb]: Since you're using this line in two methods, you may consider pulling it into it's own method.

File: lib/network.rb [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+      retval = {}
+      all_charachters.each{|char| retval[char.actor] = char.salary}
+      retval
+    end
```
Comment (mstang15) [inline lib/network.rb]: Great job! Something you may want to look into for a scenario like this is using the method #inject

File: lib/show.rb [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+  def initialize(name, creator, characters)
+    @name = name
+    @creator = creator
+    @characters = characters
+  end
+
+  def total_salary
+    @characters.sum{|char|char.salary}
+  end
```
Comment (mstang15) [inline lib/show.rb]: Great use of #sum

File: test/charachter_test.rb [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+
+class CharacterTest < Minitest::Test
+
+  def test_has_attributes
+    kitt = Character.new({name: "KITT", actor: "William Daniels", salary: 1_000_000})
+    assert_equal "KITT", kitt.name
+    assert_equal "William Daniels", kitt.actor
+    assert_equal 1000000, kitt.salary
+  end
+
```
Comment (mstang15) [inline test/charachter_test.rb]: Also, be sure to test that kitt is an instance of the Character class.

File: test/network_test.rb [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+    #whitespace intentional in setup for readability
+    @michael_knight = Character.new({name: "Michael Knight", actor: "David Hasselhoff", salary: 1_600_000})
+    @kitt = Character.new({name: "KITT", actor: "William Daniels", salary: 1_000_000})
+    @knight_rider = Show.new("Knight Rider", "Glen Larson", [@michael_knight, @kitt])
+
+    @leslie_knope = Character.new({name: "Leslie Knope", actor: "Amy Poehler", salary: 2_000_000})
+    @ron_swanson = Character.new({name: "Ron Swanson", actor: "Nick Offerman", salary: 1_400_000})
+    @parks_and_rec = Show.new("Parks and Recreation", "Michael Shur & Greg Daniels", [@leslie_knope, @ron_swanson])
+  end
+
```
Comment (mstang15) [inline test/network_test.rb]: Same thing here, just test the instance of the class first

File: test/network_test.rb [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+  end
+  #got the wierd problem ruby has with assert_equal thinking its a block, parens didnt fix
+  def test_makes_payroll
+    @nbc.add_show(@knight_rider)
+    @nbc.add_show(@parks_and_rec)
+    hash = {"David Hasselhoff" => 1600000, "William Daniels" => 1000000, "Amy Poehler" => 2000000, "Nick Offerman" => 1400000}
+    assert_equal hash,  @nbc.payroll
+  end
+
+end
```
Comment (mstang15) [inline test/network_test.rb]: 💯

File: test/show_test.rb [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+    assert_equal "Glen Larson", @knight_rider.creator
+    assert_equal [@michael_knight, @kitt], @knight_rider.characters
+  end
+
+  def test_total_salaries_of_charachters
+    assert_equal 2600000, @knight_rider.total_salary
+  end
+
+
+end
```
Comment (mstang15) [inline test/show_test.rb]: 💯

**Comment by Patrick-Duvall** [comment]
Makes sense, Reduce worked best in that one case.

**Comment by mstang15** [comment]
@Patrick-Duvall Yup! inject and reduce are aliases :)

**Pull request closed without merging**